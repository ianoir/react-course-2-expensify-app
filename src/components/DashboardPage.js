// Set up the imports for the React Components
import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';

const DashboardPage = () => (
    <div>
        <ExpenseListFilters />    
        <ExpenseList />
    </div>
);

export default DashboardPage;