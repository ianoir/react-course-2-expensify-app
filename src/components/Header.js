// Set up the imports for the React Components
import React from 'react';
import { NavLink } from 'react-router-dom';
import { connect } from 'react-redux';
import { startLogout } from '../actions/auth';

export const Header = ({ startLogout }) => (
    <header>
        <h1>Expensify</h1>
        <NavLink to="/dashboard" activeClassName="isActive">Dashboard</NavLink>
        <NavLink to="/create" activeClassName="isActive">Create an new expense</NavLink>
        <button onClick={startLogout}>Log out</button>
    </header>
);

const mapDispatchToProps = (dispatch) => ({
    startLogout: () => dispatch(startLogout())
});

export default connect(undefined, mapDispatchToProps)(Header);