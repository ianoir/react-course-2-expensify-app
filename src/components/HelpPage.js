// Set up the imports for the React Components
import React from 'react';

const HelpPage = () => (
    <div>This is from my helpcomponent</div>
);

export default HelpPage;