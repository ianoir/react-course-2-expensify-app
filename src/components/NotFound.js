// Set up the imports for the React Components
import React from 'react';
import { Link } from 'react-router-dom';

const NotFoundPage = () => (
    <div>
        Error 404! - <Link to ="/">Go to Home</Link>
    </div>
);

export default NotFoundPage;