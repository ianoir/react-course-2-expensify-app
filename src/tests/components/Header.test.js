import React from 'react';
import { shallow } from 'enzyme';
import { Header } from '../../components/Header';

// Should render header correctly
test('Should render header correctly', () => {
    const wrapper = shallow(<Header startLogout={() => {}} />);
    expect(wrapper).toMatchSnapshot();
});

// Should call startLogout on button click
test('Should call startLogin on button click', () => {
    const logoutSpy = jest.fn();
    const wrapper = shallow(<Header startLogout={logoutSpy} />);
    wrapper.find('button').simulate('click');

    expect(logoutSpy).toHaveBeenCalled();
});