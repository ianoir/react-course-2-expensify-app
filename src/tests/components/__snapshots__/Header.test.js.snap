// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="isActive"
    ariaCurrent="true"
    to="/dashboard"
  >
    Dashboard
  </NavLink>
  <NavLink
    activeClassName="isActive"
    ariaCurrent="true"
    to="/create"
  >
    Create an new expense
  </NavLink>
  <button
    onClick={[Function]}
  >
    Log out
  </button>
</header>
`;
